ocaml/typing/env.ml

```diff
@@ -3094,7 +3094,10 @@ let unboxed_type ~errors ~env ~loc ~lid ty =
   match ty with
   | None -> ()
   | Some ty ->
-    match !constrain_type_jkind env ty Jkind.Primitive.(value ~why:Captured_in_object) with
+    (* Sometimes, this function is called on a generalized type variable
+       when it actually should check the specialized one. This is sound,
+       but incomplete. *)
```

Comment:
```suggestion
    (* The type is the type of a variable in the environment. It thus is likely generic. Despite
       the fact that instantiated variables work better in [constrain_type_jkind] (because they
       can be assigned more specific jkinds), we actually want to work on these generic types
       here. After all, it's the value in the environment that is getting captured by the object,
       not a specific instance of that variable. *)
```